### ISSUE TITLE
Boost.Python.ArgumentError: Python argument types in     FragCatGenerator.AddFragsFromMol(FragCatGenerator, NoneType, FragCatalog) did not match C++ signature:     AddFragsFromMol(RDKit::FragCatGenerator {lvalue}, RDKit::ROMol, RDCatalog::HierarchCatalog<RDKit::FragCatalogEntry, RDKit::FragCatParams, int>*)

### ISSUE THREAD

**Issue by Vencent-Won** [rdkit/rdkit#5323]
-----------------------------------------

If you have a question about the RDKit or aren't sure if what you're seeing is the right behavior please use the Discussions tab above (https://github.com/rdkit/rdkit/discussions) instead of posting it here.

Please help us help you: We normally need the information below in order to answer questions or understand bug reports. If you do not provide the information requested we may not be able to help you and will probably close the issue.

PLEASE DELETE THIS SECTION AFTER YOU HAVE READ IT

-----------------------------------------

**Describe the bug**
Boost.Python.ArgumentError: Python argument types in     FragCatGenerator.AddFragsFromMol(FragCatGenerator, NoneType, FragCatalog) did not match C++ signature:     AddFragsFromMol(RDKit::FragCatGenerator {lvalue}, RDKit::ROMol, RDCatalog::HierarchCatalog<RDKit::FragCatalogEntry, RDKit::FragCatParams, int>*)

**To Reproduce**
mol = 'O=N(=O)C1=CC2=N(N=CC2C=C1)C'
fName = os.path.join(RDConfig.RDDataDir, 'FunctionalGroups.txt')
fcat = FragmentCatalog.FragCatalog(fparams)
fcgen = FragmentCatalog.FragCatGenerator()
mol = Chem.MolFromSmiles(mol)
fparams = FragmentCatalog.FragCatParams(1, 6, fName)
fcgen.AddFragsFromMol(mol, fact)

Then the error occured.

**Expected behavior**
I hope that the mol can be fragmented. Then I can get the groups of this mol. 

**Screenshots**

**Configuration (please complete the following information):**
 - RDKit version:
 -2022.03.02 
 - OS: [e.g. Ubuntu 20.04]
-16.0.4 
- Python version (if relevant):
-3.7 
- Are you using conda?
 - yes.
 - If you are using conda, which channel did you install the rdkit from?
 - conda install rdkit rdkit-dev
 - If you are not using conda: how did you install the RDKit?

**Additional context**

**Comment by bp-kelley** [comment]
You have a hypervalent Nitrogen there.

```
>>> Chem.MolFromSmiles('O=N(=O)C1=CC2=N(N=CC2C=C1)C')
[12:49:40] Explicit valence for atom # 6 N, 4, is greater than permitted
```

this returns None as a molecule which causes your error in the call to AddFragsFromMol

**Comment by greglandrum** [comment]
> So, in rdkit, the Nitrogen is only 3 value ?

Neutral nitrogen should have three bonds. That's basic chemistry of stable molecules.

The RDKit has rules for automatically correcting the valence of things like nitro groups, but the four-coordinate neutral nitrogen in your molecule is very likely wrong.

**Comment by Vencent-Won** [comment]
You are right, then may I ask if there is a function for correcting the valence of molecules in the rdkit package?

**Comment by Vencent-Won** [comment]
You are right, then may I ask if there is a function for correcting the valence or bonds of molecules in the rdkit package?

**Comment by greglandrum** [comment]
I'm closing this because it is a question, not a bug report or enhancement request.
Please ask questions using the "Discussions" tab.
